src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrToReturn = [];
+  // copying state
+  let stateCopy = { ...state };
+
+  // iterate throught actions array
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    if (type === 'addProperties') {
+      // Add to already existing state extraData.
+      stateCopy = { ...stateCopy, ...extraData };
+    }
+
+    if (type === 'removeProperties') {
+      // This one is easy to understand actually
+      for (const key of keysToRemove) {
+        delete stateCopy[key];
+      }
+    }
+
+    // Reset stateCopy, like IPhone
+    if (type === 'clear') {
+      stateCopy = {};
+    }
```

Comment:
[CODE STYLE]: Use switch statement if you have limited amount of conditions. This will make your code cleaner and more organized. Convert the `if` statements to a `switch` statement to handle the action types.